lighthouse-core/audits/installable-manifest.js

```diff
@@ -51,59 +125,108 @@ class InstallableManifest extends MultiCheckAudit {
   }
 
   /**
-   * @param {LH.Artifacts.ManifestValues} manifestValues
-   * @return {Array<string>}
+   * @param {LH.Artifacts} artifacts
+   * @return {{i18nErrors: Array<LH.IcuMessage | string>; warnings: Array<LH.IcuMessage>}}
    */
-  static assessManifest(manifestValues) {
-    if (manifestValues.isParseFailure && manifestValues.parseFailureReason) {
-      return [manifestValues.parseFailureReason];
+  static getInstallabilityErrors(artifacts) {
+    const installabilityErrors = artifacts.InstallabilityErrors.errors;
+    const i18nErrors = [];
+    const warnings = [];
+    const errorArgumentsRegex = /{([^}]+)}/g;
+
+    for (const err of installabilityErrors) {
+      // Filter out errorId 'in-incognito' since Lighthouse recommends incognito.
+      if (err.errorId === 'in-incognito') continue;
+
+      if (err.errorId === 'warn-not-offline-capable') {
+        warnings.push(str_(UIStrings[err.errorId]));
+        continue;
+      }
+
+      // @ts-expect-error errorIds from protocol should match up against the strings dict
+      const matchingString = UIStrings[err.errorId];
+
+      if (matchingString === undefined) {
+        i18nErrors.push(str_(UIStrings.noErrorId, {errorId: err.errorId}));
+        continue;
+      }
+
+      // Get the arguments of the installability error message.
+      const UIStringArguments = matchingString.match(errorArgumentsRegex) || [];
+
+      /**
+       * If there is an argument value, get it.
+       * We only expect a `minimum-icon-size-in-pixels` errorArg[0] for two errorIds, currently.
+       */
+      const value0 = err.errorArguments && err.errorArguments.length && err.errorArguments[0].value;
+
+      if (matchingString && err.errorArguments.length !== UIStringArguments.length) {
+        // Matching string, but have the incorrect number of arguments for the message.
+        const argsReceived = err.errorArguments.join(', ');
+        const msg = err.errorArguments.length > UIStringArguments.length ?
+          ` has unexpected arguments {${argsReceived}}` :
```

Comment:
add a test for this case, too (not just the "does not have the expected number of arguments" one)?